cmd/flexible-informer-gen/generators/informer.go

```diff
@@ -163,6 +165,10 @@ func NewFiltered$.type|public$Informer(client $.clientSetInterface|raw$$if .name
 				unstructuredResult := &unstructured.UnstructuredList{}
 				err := client.$.group$$.version$().RESTClient().Get().$if .namespaced$Namespace(namespace).$end$Resource($.toLower$("$.type|publicPlural$")).VersionedParams(&options, $.schemes|raw$).Timeout(timeout).Do(context.TODO()).Into(unstructuredResult)
 				if err != nil {
+					if $.isNotFound$(err) {
+						logs.warnf("The version of Istio CRD is not matched with Aspen Mesh", logs.ErrorField(err))
+						return result, nil
```

Comment:
With regards to the logging statement: I think it makes sense here to use both the CRD and the version that we were expecting but unable to find (by using `$.version$` and `$.type|raw$`, I think) to help end users debug what the issue is.

With regards to the return statement, what is the desired effect here? Do we ultimately want the controlplane to come up but be unable to list things? I'm not sure returning the error is correct, either, since this would just change the log error while still having the exact same symptoms as the bug we're trying to fix. I'm not quite sure that panicking is the right approach, either.

Reply:
I think controlplane will list an empty list of the resources as it's 'not found'. Your suggestions about the logging make sense.

Reply:
That's my understanding as well (I was imprecise when I said "unable to list anything"). I'm guessing this is the conclusion you already came to with Andrew, so I'm happy with that.

Reply:
Yeah I think we decided this one should allow it to come up anyway.

If you want a fail-y informer, you should use the "strict" informer (and panic if the caches don't sync).  I know that's not implemented yet but that's my thought.

If we need more flexibility, we could allow you to pass in an error handling callback.  I do like that idea as that callback could do things like tick prometheus metrics or add warnings to some API that our UI could check.  Probably future though.

Reply:
I think assuming we're in agreement on the behavior (don't panic - this is the flexible informer), then both this convo and Tom's thread are about clarity in the logging and it sounds like we may all be in agreement there too.